development/components/database/objectmodel.md

```diff
@@ -64,86 +104,443 @@ public static $definition = [
+- `validate` is a validation rule (optional)
+- `size` is the max size of the field. It should match the database definition to avoid data cropping or overflow when inserting/updating objects.
+
+#### Field types reference
+
+Field type is an important setting, it determines how ObjectModel will format your data.
+
+<table>
+    <thead>
+        <tr>
+            <th>Type in ObjectModel</th>
+            <th>Related type in MySQL</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>TYPE_INT</td>
+            <td>INTEGER, INT, SMALLINT, TINYINT, MEDIUMINT, BIGINT, ...</td>
+        </tr>
+        <tr>
+            <td>TYPE_BOOL</td>
+            <td>SMALLINT</td>
+        </tr>
+        <tr>
+            <td>TYPE_STRING</td>
+            <td>VARCHAR</td>
+        </tr>
+        <tr>
+            <td>TYPE_FLOAT</td>
+            <td>FLOAT, DOUBLE</td>
+        </tr>
+        <tr>
+            <td>TYPE_DATE</td>
+            <td>DATE</td>
+        </tr>
+        <tr>
+            <td>TYPE_HTML</td>
+            <td>BLOB, TEXT</td>
+        </tr>
+        <tr>
+            <td>TYPE_NOTHING</td>
+            <td>Should not be used (avoids ObjectModel formating)</td>
+        </tr>
+        <tr>
+            <td>TYPE_SQL</td>
+            <td>Should not be used</td>
+        </tr>
+    </tbody>
+</table>
+
+#### Validation rules reference
+
+Several validation rules are available for your ObjectModel fields. 
+[Please refer to the Validate class of PrestaShop](https://github.com/PrestaShop/PrestaShop/blob/8.0.x/classes/Validate.php) for a complete list.
+
+## Basic usage of an ObjectModel managed entity
+
+### Create and save a new object
+
+```php
+$cms = new Cms();
+$cms->position = 2;
+...
+$cms->save();
+```
+
+In this example, we create an entity from scratch. Then, we set its `position` attribute, and we call the `save()` method. The `save()` method will trigger the `add()` method since its `id` attribute is not yet known (because the entity is not created in database).
+
+If the insert was successful, the ObjectModel class will set the id of the entity (retrieved from database). [Complete reference here](https://github.com/PrestaShop/PrestaShop/blob/8.0.x/classes/ObjectModel.php#L572-L658).
+
 
-In order to retrieve an object in many languages:
+### Load and save an object
 
 ```php
-'multilang' => true
+$id = 2; // id of the object in database
+$cms = new Cms($id); 
+$cms->position = 3;
+...
+$cms->save();
 ```
 
-In order to retrieve an object depending on the current store:
+In this example, we retrieve an entity from the database, with its id. Then, we change its `position` attribute, and we call the same `save()` method. The `save()` method will trigger the `update()` method and not the `add()` method since its `id` attribute is known. [Complete reference here](https://github.com/PrestaShop/PrestaShop/blob/8.0.x/classes/ObjectModel.php#L750-L868).
```

Comment:
```suggestion
In this example, we retrieve an entity from the database with its id. Then, we change its `position` attribute and call the same `save()` method. The `save()` method will trigger the `update()` method and not the `add()` method since its `id` attribute is known. [Complete reference here](https://github.com/PrestaShop/PrestaShop/blob/8.0.x/classes/ObjectModel.php#L750-L868).
```